dataprofiler/profilers/helpers/report_helpers.py

```diff
@@ -41,37 +41,105 @@ def flat_dict(od, separator='_', key=''):
            } if isinstance(od, dict) else {key:od}
 
 
-def _prepare_report(report, output_format=None):
-    if output_format:
+def _prepare_report(report, output_format=None, omit_keys=[]):
+    """
+    Prepares report dictionary for users upon request.
+
+    :param report: contains the values identified from the profile
+    :type report: dict()
+    :param output_format: designation for how to format the returned report
+    :type output_format: dict()
+    :param omit_keys: Keys to omit from the output report, to omit keys in the 
+                      report a '.' represents a level of recursion example: 
+                      report: { 'test1': { 'test2': val, 'test3': val }, 
+                      to omit key 'test3' from report: omit_keys=['test1.test3']
+                      wildcards are also possible, so: omit_keys=['*.test3']
+    :type omit_keys: list(str)
+    :return report: handle to the updated report
+    :type report: dict()
+    """
+    
+    format_options = ['pretty', 'serializable', 'flat', 'compact']    
+    if output_format is not None:
         output_format = output_format.lower()
-    if not output_format or output_format not in ['pretty', 'serializable', 'flat']:
-        return report
-    report = report.copy()
+    if omit_keys is None:
+        omit_keys = []
+
+    fmt_report = {}
     max_str_len = 50
     max_array_len = 5
+
+    if output_format == 'compact':
+        omit_keys.extend([
+            "data_stats.*.statistics.times",
+            "data_stats.*.statistics.avg_predictions",
+            "data_stats.*.statistics.data_label_representation",
+            "data_stats.*.statistics.null_types_index",
+            "data_stats.*.statistics.histogram.bin_counts",
+            "data_stats.*.statistics.histogram.bin_edges"
```

Comment:
did we decide to remove histogram results as well?

Reply:
The histogram result was already truncated and inaccessible via the "pretty" format. 

I simply deleted it for the "compact" format. The reality is that, if a user wants to use the histogram they'll need to use none of the format options.